tests/e2e/e2e_test.go

```diff
@@ -476,3 +513,325 @@ var _ = Describe("odoe2e", func() {
 		})
 	})
 })
+
+var _ = Describe("updateE2e", func() {
+	var t = strconv.FormatInt(time.Now().Unix(), 10)
+	var projName = fmt.Sprintf("odo-%s", t)
+	const appTestName = "testing"
+
+	const bootStrapSupervisorURI = "https://github.com/kadel/bootstrap-supervisored-s2i"
+	const initContainerName = "copy-files-to-volume"
+	const wildflyUri1 = "https://github.com/marekjelen/katacoda-odo-backend"
+	const wildflyUri2 = "https://github.com/mik-dass/katacoda-odo-backend"
+	const appRootVolumeName = "-testing-s2idata"
+
+	tmpDir, err := ioutil.TempDir("", "odo")
+	if err != nil {
+		Fail(err.Error())
+	}
+
+	Describe("creating the project", func() {
+		Context("odo project", func() {
+			It("should create a new project", func() {
+				session := runCmd("odo project create " + projName + "-1")
+				Expect(session).To(ContainSubstring(projName))
+			})
+
+			It("should get the project", func() {
+				getProj := runCmd("odo project get --short")
+				Expect(strings.TrimSpace(getProj)).To(Equal(projName + "-1"))
+			})
+		})
+	})
+
+	Describe("creating an application", func() {
+		Context("when application by the same name doesn't exist", func() {
+			It("should create an application", func() {
+				appName := runCmd("odo app create " + appTestName)
+				Expect(appName).To(ContainSubstring(appTestName))
+			})
+		})
+	})
+
+	Context("updating the component", func() {
+		It("should be able to create binary component", func() {
+			runCmd("curl -o " + tmpDir + "/sample-binary-testing-1.war " +
+				"https://gist.github.com/mik-dass/f95bd818ddba508ff76a386f8d984909/raw/e5bc575ac8b14ba2b23d66b5cb4873657e1a1489/sample.war")
```

Comment:
I was thinking if there can be any alternative to using personal gists?

Reply:
I tried to look for online war hosted files but didn't find many reliable ones. If you find any online hosted ones, let me know, I will change them to those

Reply:
Hmm... You can search for the other one ;)
The only reason why we should not depend on external links is, if they ever remove it or if the gists get auto removed due to the expiry of the assigned lifespan for the gist, our tests will start raising alarms :(
Optionally(the best way in my opinion) we could also keep them in our repo itself in a separate directory(may be samples directory for  example) and use it by way of relative path or a git url to the file(the way you want for the test)...
Not sure how high on priority this task would be... but considering the fact that maintaining them in a dir on github is a not so difficult task, we may consider giving it a try... But I would take @kadel or @cdrage opinion before doing this...

Reply:
Hmm... You can search for the other one ;)
The only reason why we should not depend on external links is, if they ever remove it or if the gists get auto removed due to the expiry of the assigned lifespan for the gist, our tests will start raising alarms :(
Optionally(the best way in my opinion) we could also keep them in our repo itself in a separate directory(may be samples directory for  example) and use it by way of relative path or a git url to the file(the way you want for the test)...
Not sure how high on priority this task would be... but considering the fact that maintaining them in a dir on github is a not so difficult task, we may consider giving it a try... But I would take @kadel or @cdrage opinion before doing this...

Reply:
If you choose to maintain them as part of our repo, consider the discussion on https://github.com/redhat-developer/odo/issues/777